src/mudlet.cpp

```diff
@@ -1404,16 +1402,17 @@ void mudlet::slot_close_profile_requested(int tab)
 
     pH->stopAllTriggers();
     pH->mpEditorDialog->close();
-    for (auto consoleName : hostConsoleMap.keys()) {
-        hostConsoleMap[consoleName]->close();
-        hostConsoleMap.remove(consoleName);
 
+    for (auto consoleName : hostConsoleMap.keys()) {
         if (dockWindowMap.contains(consoleName)) {
             dockWindowMap[consoleName]->setAttribute(Qt::WA_DeleteOnClose);
             dockWindowMap[consoleName]->close();
             removeDockWidget(dockWindowMap[consoleName]);
             dockWindowMap.remove(consoleName);
         }
+
+        hostConsoleMap[consoleName]->close();
```

Comment:
Surely calling `QDockWidget::close()` when the `Qt::WA_DeleteOnClose` property is set on it will automagically destroy the `TConsole` instance it contains {and if not, why not?} - so shouldn't the `hostConsoleMap[consoleName]->close();` be in an `else` to the `if` - though the `hostConsoleMap.remove(consoleName);` should remain outside of the `if` ? 🤓

Reply:
The entire reason for this bug is because the `TConsole` is not destroyed automatically when the `QDockWidget` is, so this solution works well.